src/initialization/Warnings.cpp

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2022 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Marco Garten
+ * License: BSD-3-Clause-LBNL
+ */
+#include "ImpactX.H"
+
+#include <ablastr/warn_manager/WarnManager.H>
+#include <ablastr/utils/TextMsg.H>
+
+#include <AMReX_BLProfiler.H>
+
+
+void impactx::ImpactX::warn_logger_control ()
+{
+    amrex::ParmParse pp_impactx("impactx");
+
+    // Set the flag to control if ImpactX has to emit a warning message as soon as a warning is recorded
+    bool always_warn_immediately = false;
+    pp_impactx.query("always_warn_immediately", always_warn_immediately);
+    ablastr::warn_manager::GetWMInstance().SetAlwaysWarnImmediately(always_warn_immediately);
+
+    // Set the WarnPriority threshold to decide if ImpactX has to abort when a warning is recorded
+    if(std::string str_abort_on_warning_threshold = "";
+            pp_impactx.query("abort_on_warning_threshold", str_abort_on_warning_threshold)){
+        std::optional<ablastr::warn_manager::WarnPriority> abort_on_warning_threshold = std::nullopt;
+        if (str_abort_on_warning_threshold == "high")
+            abort_on_warning_threshold = ablastr::warn_manager::WarnPriority::high;
+        else if (str_abort_on_warning_threshold == "medium" )
+            abort_on_warning_threshold = ablastr::warn_manager::WarnPriority::medium;
+        else if (str_abort_on_warning_threshold == "low")
+            abort_on_warning_threshold = ablastr::warn_manager::WarnPriority::low;
+        else {
+            amrex::Abort(ablastr::utils::TextMsg::Err(str_abort_on_warning_threshold
```

Comment:
Please throw a `std::runtime` error instead of calling `amrex::Abort` here.
We want to be able to catch this in all cases.